### PR TITLE
Issue #1418: wire up grant & revoke

### PR DESCRIPTION
**Pull request by dgnorton**

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -2020,6 +2020,70 @@ func (s *Server) executeShowTagValuesStatement(stmt *influxql.ShowTagValuesState
...
+		if stmt.Privilege != influxql.AllPrivileges {
+			return &Result{
+				Err: fmt.Errorf("cannot grant %s on the cluser, only %s",
+					stmt.Privilege.String(),
+					influxql.AllPrivileges.String()),
+			}
+		}
+
+		// Grant user cluster admin privileges.
+		u.Admin = true
```
Comment (otoolep) [inline server.go]: I might be issuing something, but this doesn't do anything to update the cluster. It simply updates the in-RAM user object, right? Restart this node, or hit a different node, and this change is not there.

**Comment by dgnorton** [comment]
Doh!  You're right.  Good catch.

File: httpd/handler_test.go [hunk 2]
```diff
@@ -848,6 +848,140 @@ func TestHandler_AuthenticatedDatabases_UnauthorizedBasicAuth(t *testing.T) {
...
+	if status != http.StatusOK {
+		t.Fatalf("unexpected status: %d", status)
+	}
+
+	if u := srvr.User("john"); !u.Admin {
+		t.Fatal(`expected user "john" to be admin`)
+	}
+
+	// Make sure update persists after server restart.
+	srvr.Restart()
```
Comment (otoolep) [inline httpd/handler_test.go]: Nice.

**Comment by otoolep** [comment]
LGTM. Perhaps a unit test testing behavior when the database doesn't exist?

**Comment by dgnorton** [comment]
I left it like MySQl, so a user can be granted privileges on a db that doesn't exist.  The code doesn't look up the database during grant / revoke, so there isn't really anything to test.

**Pull request merged**